docs/how-to/llm-fine-tuning-optimization/model-acceleration-libraries.rst

```diff
@@ -251,3 +251,265 @@ page describes the options.
+and as a back end for PyTorch quantized operators. FBGEMM offers optimized on-CPU performance for reduced precision calculations,
+strong performance on native tensor formats, and the ability to generate
+high-performance shape- and size-specific kernels at runtime.
+
+Meta now supports FBGEMM on ROCm as part of the FBGEMM_GPU GPU kernel library. FBGEMM_GPU
+collects several high-performance PyTorch GPU operator libraries  
+for use in training and inference. It provides efficient table-batched embedding functionality,
+data layout transformation, and quantization support.
+
+For more information about FBGEMM, see the `PyTorch FBGEMM GitHub <https://github.com/pytorch/FBGEMM>`_
+and the `PyTorch FBGEMM documentation <https://pytorch.org/FBGEMM/>`_.
+The `Meta blog post about FBGEMM <https://engineering.fb.com/2018/11/07/ml-applications/fbgemm/>`_ provides
+additional background about the library.
+
+Installing FBGEMM 
+----------------------
+
+Installing FBGEMM and FBGEMM_GPU consists of the following steps:
+
+*  Set up an isolated Miniconda environment
+*  Install any necessary ROCm components
+*  Install the build tools
+*  Install `PyTorch <https://pytorch.org/>`_
+*  Complete the pre-build and build tasks
+
+Set up the Miniconda environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To install Miniconda, use the following commands.
+
+#. Install a `Miniconda environment <https://docs.anaconda.com/miniconda/>`_ for reproducible builds.
+   All subsequent commands run inside this environment.
+
+   .. code-block:: shell
+
+      export PLATFORM_NAME="$(uname -s)-$(uname -m)"
+
+      # Set the Miniconda prefix directory
+      miniconda_prefix=$HOME/miniconda
+
+      # Download the Miniconda installer
+      wget -q "https://repo.anaconda.com/miniconda/Miniconda3-latest-${PLATFORM_NAME}.sh" -O miniconda.sh
+
+      # Run the installer
+      bash miniconda.sh -b -p "$miniconda_prefix" -u
+
+      # Load the shortcuts
+      . ~/.bashrc
+
+      # Run updates
+      conda update -n base -c defaults -y conda
+
+#. Create a Miniconda environment with Python version 3.12:
+
+   .. code-block:: shell
+
+      env_name=<ENV NAME>
+      python_version=3.12
+
+      # Create the environment
+      conda create -y --name ${env_name} python="${python_version}"
+
+      # Upgrade PIP and pyOpenSSL package
+      conda run -n ${env_name} pip install --upgrade pip
+      conda run -n ${env_name} python -m pip install pyOpenSSL>22.1.0
+
+Install the ROCm components
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+FBGEMM requires installation of the full ROCm package. It also requires the :doc:`MIOpen <miopen:index>` component
+as a dependency. To install MIOpen, use the ``apt install`` command.
+
+.. code-block:: shell
+
+   apt install hipify-clang miopen-hip miopen-hip-dev
+
+Install the build tools
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Install the GCC compiler, create the compiler symlinks, and install some additional build tools.
```

Comment:
> (In my experience the rocm clang++ is good enough for building fbgemm_gpu, we may delete this sub-section, please request other reviewers' comments @naromero77amd).

The default behavior of FBGEMM is that it will use GCC on the host and HIP-enabled clang++ for the GPU. Most of the AMD docker images will provide a GCC paired up with a HIP-enabled clang++. So most users are unlikely to be installing their own GCC.

Reply:
I will remove this section. I'll move the instruction about the additional build tools up to the first section (because it doesn't make sense to have a separate section for one minor command).